playbooks/init/facts.yml

```diff
@@ -100,3 +100,5 @@
       # We need to setup openshift_client_binary here for special uses of delegate_to in
       # later roles and plays.
       first_master_client_binary: "{{  openshift_client_binary }}"
+      #Some roles may require this to be set for first master
+      openshift_client_binary: "{{ openshift_client_binary }}"
```

Comment:
This should be unnecessary if the proper role is in scope.

first_master_client_binary is used because of delegate_to.  delegate_to doesn't seem to resolve the variable's value on the delegated host, just on the inventory_hostname.  Thus, in situations where the master may have a different oc binary value than the node, we need to set this as a fact.

It's not a good practice to set a fact to a value that may potentially be user provided.  In this particular case, it doesn't look like it will cause any obvious issues.

All in all, I prefer 'first_master_client_binary' as-is due to easier finding with grep.

Reply:
This is failing in upgrade playbooks because we've supplied a list of tasks in the playbooks rather than calling any role that would've scoped in openshift_facts.
While an admin could set openshift_client_binary we've never documented it and it's only ever been computed based on whether or not it's a containerized installation (/usr/local/bin/oc versus oc). This sure feels like we should be setting this in init so it's always available.